### DISCUSSION TITLE
Running behind Traefik Reverse Proxy

### DISCUSSION THREAD

**Discussion by chandanvp** [nextcloud/all-in-one#543] [Questions] [answered]
Hi,
I tried out Nextcloud AIO on docker and it is really AIO stop for Nextcloud. I used the official docker-compose file to run AIO and also start containers as well. 
But I want to run AIO behind Traefik Reverse Proxy. Traefik uses docker labels to route traffic and provides SSL as well. It runs on 80,443 and I tried adding 8080 port in label and then access it using URL. It worked but, as soon as I login, I get an error saying "Slim Application Error" and nothing shows up. I am not sure how to proceed further with Traefik or AIO. If anyone got it to working using Traefik, I can have a check at your compose file for better understanding.

Also, another question is, How to change the volume mount from a docker-volume to a host path mount? I added host path instead of docker volume, but container didn't even start and keep showing 

"nextcloud-aio  | It seems like you did not give the mastercontainer volume the correct name?"

**Comment by szaimen** [comment]
Hello, I currently see no way around using a traefik.toml if you want to run Nextcloud AIO behind Traefik (also see the network host option in the reverse proxy documentation). But I cannot help further here as I don't run Traefik myself. After someone figured this out I'll gladly add it to the reverse-proxy documentation.

---

> Also, another question is, How to change the volume mount from a docker-volume to a host path mount? I added host path instead of docker volume, but container didn't even start and keep showing

There are these two options: https://github.com/nextcloud/all-in-one#how-to-store-the-filesinstallation-on-a-separate-drive and https://github.com/nextcloud/all-in-one#how-to-change-the-default-location-of-nextclouds-datadir

**Reply by szaimen** [reply]
You can also see the reply in https://github.com/nextcloud/all-in-one/discussions/553#discussioncomment-2679936

**Comment by macaussie** [comment]
I used docker-compose and added the apache port as shown below.



  nextcloud:
    image: nextcloud/all-in-one:latest
    restart: unless-stopped
    container_name: nextcloud-aio-mastercontainer
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config
      - /var/run/docker.sock:/var/run/docker.sock:ro
    ports:
      - 16000:80
      - 16001:8080
      - 16002:8443
    environment:
      - APACHE_PORT=11000
volumes:
  nextcloud_aio_mastercontainer:
    name: nextcloud_aio_mastercontainer



Then in the traefik rules folder I created nc.toml with the following config



[http.routers]
  [http.routers.nc-rtr]
      entryPoints = ["https"]
      rule = "Host(`nc.mydomain.com`)"
      service = "nc-svc"
      middlewares = ["chain-no-auth"]
      [http.routers.nc-rtr.tls]
        certresolver = "dns-cloudflare"

[http.services]
  [http.services.nc-svc]
    [http.services.nc-svc.loadBalancer]
      passHostHeader = true
      [[http.services.nc-svc.loadBalancer.servers]]
        url = "http://xxx.xxx.xxx.xxx:11000"



Now I have Nextcloud behind a Traefik2 (docker) reverse proxy.

I hope this helps.

**Reply by szaimen** [reply]
Hello, thank you very much for this! :)
Would you be fine if I add this to the reverse proxy documentation and what would be different when using lets encrypt and not dns-cloudflare as certresolver?

**Reply by macaussie** [reply]
Yes feel free.

My certs are from Let's Encrypt, however Cloudflare hosts my domain and is heavily referenced through my docker-compose file for other services.

**Reply by szaimen** [reply]
Great, thank you! 

So I suppose the config above will only work for users that host their domain on cloudflare? What would need to be changed to make it work with others?

**Reply by macaussie** [reply]
Most of that is relevant to Traefik. 

Looking at the Traefik docs, it seems you can also use Lets Encrypt

traefik.http.routers.blog.tls.certresolver=le

**Reply by szaimen** [reply]
okay cool! I'll will create a PR then :)

**Reply by robotictacos** [reply]
This solution is using the outdated version of Traefik.  I'm on the latest version of Traefik, and here is my config:

   nextcloud:
    image: nextcloud/all-in-one:latest
    container_name: nextcloud-aio-mastercontainer
    environment:
      - APACHE_PORT=11000
    networks:
      t2_proxy:
        ipv4_address: 172.18.0.16
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config
      - $DOCKERDIR/nextcloud/data:/data
      - /var/run/docker.sock:/var/run/docker.sock:ro
    ports:
      - 3478:3478
    restart: unless-stopped
    labels:
      - "traefik.enable=true"
      ## HTTP Routers
      - "traefik.http.routers.nextcloud-rtr.entrypoints=https"
      - "traefik.http.routers.nextcloud-rtr.rule=Host(`nextcloud.$DOMAINNAME`)"
      ## Middlewares
      - "traefik.http.routers.nextcloud-rtr.middlewares=chain-no-auth@file"
      ## HTTP Services
      - "traefik.http.routers.nextcloud-rtr.service=nextcloud-svc"
      - "traefik.http.services.nextcloud-svc.loadbalancer.server.port=8080"

I think this is correct (although as OP points out, would prefer to use host mapped folders instead of volumes so I can view/update config), but I get an error when I access the UI:

Bad Request
Your browser sent a request that this server could not understand.
Reason: You're speaking plain HTTP to an SSL-enabled server port.
Instead use the HTTPS scheme to access this URL, please.

Makes sense.  Is there a port exposed that speaks plain http that I can proxy to, since 8080 is encrypted?  I have no need for the container to encrypt itself since Traefik handles all that for me (for all my other dockers as well).  I tried port 80 and 11000 before replying but no joy.

**Reply by szaimen** [reply]
Hello @robotictacos, please have a look at this: https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md and this https://github.com/nextcloud/all-in-one/discussions/543#discussioncomment-2669959

**Reply by szaimen** [reply]
Treafik 2 support will be documented with https://github.com/nextcloud/all-in-one/pull/564

**Reply by macaussie** [reply]
I am using Traefik v2.2.11 (chevrotin)

**Reply by szaimen** [reply]
@macaussie do you mind having a fast look at my PR if that is fine? :)

**Reply by robotictacos** [reply]
@szaimen I read both.  I think I know what's going on, not sure how to fix it though.  I can get the AIO setup screen to start the setup process, but only by using @macaussie 's suggestion of mapping ports 16000-16002 to the internal ports of 80, 8080 and 8443, and then navigating to https://myurl:16001.  So progress!  So it could be that the 11000 port doesn't actually go anywhere until after the initial setup (can you confirm that?).  Otherwise I get a Bad Gateway error.

Now I am stuck on setup.  It's looking for a borg backup file, which I don't have because this is a new install.  Any tips to get past that?

**Reply by szaimen** [reply]
you need to enter the domain that you want to use

**Reply by robotictacos** [reply]
I entered the domain, that part is fine, but I don't have a backup archive, and the setup will not proceed without one.

**Reply by szaimen** [reply]
don't you get any error message?

**Reply by szaimen** [reply]
Maybe javascript is disabled?

**Comment by robotictacos** [comment]
Here is the error shown in the container, note that I tried to provide a (existing) folder and password, in case it wants to create a new backup:

[Wed May 04 16:55:09.695316 2022] [php:notice] [pid 654] [client 127.0.0.1:43692] Slim Application Error\nType: GuzzleHttp\\Exception\\ServerException\nCode: 500\nMessage: Server error: `POST http://localhost/v1.41/containers/nextcloud-aio-borgbackup/start` resulted in a `500 Internal Server Error` response:\n{&quot;message&quot;:&quot;Bind mount failed: '/mnt/backup' does not exists&quot;}\n\n\nFile: /var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php\nLine: 113\nTrace: #0 /var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/Middleware.php(69): GuzzleHttp\\Exception\\RequestException::create(Object(GuzzleHttp\\Psr7\\Request), Object(GuzzleHttp\\Psr7\\Response), NULL, Array, NULL)\n#1 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(204): GuzzleHttp\\Middleware::GuzzleHttp\\{closure}(Object(GuzzleHttp\\Psr7\\Response))\n#2 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(153): GuzzleHttp\\Promise\\Promise::callHandler(1, Object(GuzzleHttp\\Psr7\\Response), NULL)\n#3 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/TaskQueue.php(48): GuzzleHttp\\Promise\\Promise::GuzzleHttp\\Promise\\{closure}()\n#4 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(248): GuzzleHttp\\Promise\\TaskQueue->run(true)\n#5 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(224): GuzzleHttp\\Promise\\Promise->invokeWaitFn()\n#6 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(269): GuzzleHttp\\Promise\\Promise->waitIfPending()\n#7 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(226): GuzzleHttp\\Promise\\Promise->invokeWaitList()\n#8 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(62): GuzzleHttp\\Promise\\Promise->waitIfPending()\n#9 /var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/Client.php(187): GuzzleHttp\\Promise\\Promise->wait()\n#10 /var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/ClientTrait.php(95): GuzzleHttp\\Client->request('POST', 'http://localhos...', Array)\n#11 /var/www/docker-aio/php/src/Docker/DockerActionManager.php(177): GuzzleHttp\\Client->post('http://localhos...')\n#12 /var/www/docker-aio/php/src/Controller/DockerController.php(50): AIO\\Docker\\DockerActionManager->StartContainer(Object(AIO\\Container\\Container))\n#13 /var/www/docker-aio/php/src/Controller/DockerController.php(125): AIO\\Controller\\DockerController->PerformRecursiveContainerStart('nextcloud-aio-b...')\n#14 /var/www/docker-aio/php/vendor/slim/slim/Slim/Handlers/Strategies/RequestResponse.php(43): AIO\\Controller\\DockerController->StartBackupContainerTest(Object(GuzzleHttp\\Psr7\\ServerRequest), Object(GuzzleHttp\\Psr7\\Response), Array)\n#15 /var/www/docker-aio/php/vendor/slim/slim/Slim/Routing/Route.php(384): Slim\\Handlers\\Strategies\\RequestResponse->__invoke(Array, Object(GuzzleHttp\\Psr7\\ServerRequest), Object(GuzzleHttp\\Psr7\\Response), Array)\n#16 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(81): Slim\\Routing\\Route->handle(Object(GuzzleHttp\\Psr7\\ServerRequest))\n#17 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(81): Slim\\MiddlewareDispatcher->handle(Object(GuzzleHttp\\Psr7\\ServerRequest))\n#18 /var/www/docker-aio/php/vendor/slim/slim/Slim/Routing/Route.php(341): Slim\\MiddlewareDispatcher->handle(Object(GuzzleHttp\\Psr7\\ServerRequest))\n#19 /var/www/docker-aio/php/vendor/slim/slim/Slim/Routing/RouteRunner.php(84): Slim\\Routing\\Route->run(Object(GuzzleHttp\\Psr7\\ServerRequest))\n#20 /var/www/docker-aio/php/vendor/slim/csrf/src/Guard.php(456): Slim\\Routing\\RouteRunner->handle(Object(GuzzleHttp\\Psr7\\ServerRequest))\n#21 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(209): Slim\\Csrf\\Guard->process(Object(GuzzleHttp\\Psr7\\ServerRequest), Object(Slim\\Routing\\RouteRunner))\n#22 /var/www/docker-aio/php/vendor/slim/twig-view/src/TwigMiddleware.php(115): Psr\\Http\\Server\\RequestHandlerInterface@anonymous->handle(Object(GuzzleHttp\\Psr7\\ServerRequest))\n#23 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(147): Slim\\Views\\TwigMiddleware->process(Object(GuzzleHttp\\Psr7\\ServerRequest), Object(Psr\\Http\\Server\\RequestHandlerInterface@anonymous))\n#24 /var/www/docker-aio/php/src/Middleware/AuthMiddleware.php(38): Psr\\Http\\Server\\RequestHandlerInterface@anonymous->handle(Object(GuzzleHttp\\Psr7\\ServerRequest))\n#25 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(313): AIO\\Middleware\\AuthMiddleware->__invoke(Object(GuzzleHttp\\Psr7\\ServerRequest), Object(Psr\\Http\\Server\\RequestHandlerInterface@anonymous))\n#26 /var/www/docker-aio/php/vendor/slim/slim/Slim/Middleware/ErrorMiddleware.php(107): Psr\\Http\\Server\\RequestHandlerInterface@anonymous->handle(Object(GuzzleHttp\\Psr7\\ServerRequest))\n#27 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(147): Slim\\Middleware\\ErrorMiddleware->process(Object(GuzzleHttp\\Psr7\\ServerRequest), Object(Psr\\Http\\Server\\RequestHandlerInterface@anonymous))\n#28 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(81): Psr\\Http\\Server\\RequestHandlerInterface@anonymous->handle(Object(GuzzleHttp\\Psr7\\ServerRequest))\n#29 /var/www/docker-aio/php/vendor/slim/slim/Slim/App.php(215): Slim\\MiddlewareDispatcher->handle(Object(GuzzleHttp\\Psr7\\ServerRequest))\n#30 /var/www/docker-aio/php/vendor/slim/slim/Slim/App.php(199): Slim\\App->handle(Object(GuzzleHttp\\Psr7\\ServerRequest))\n#31 /var/www/docker-aio/php/public/index.php(155): Slim\\App->run()\n#32 {main}, referer: https://nextcloud.mydomain.com:16001/containers

If I don't enter anything and click the submit button, I very briefly see a circle, then nothing, as if it will not proceed without *some sort* of input.

**Reply by szaimen** [reply]
so you want to restore your instance from an backup archive of a former aio instance?

**Reply by szaimen** [reply]
I have never seen this issue. does the folder /mnt exist?

**Reply by robotictacos** [reply]
No, I don't have a backup at all, I'm trying to just get the software to start.  The mnt folder exists inside the container, it's mapped to a folder on the host machine.

**Reply by szaimen** [reply]
then please don't try to restore a backup since you don't have one. You need to enter the domain that you want to use for Nextcloud

**Reply by robotictacos** [reply]
I did that, but then I cannot proceed if I don't enter anything in the backup fields.

**Reply by szaimen** [reply]
can you share a screenshot of the error?
is there an error at all when you submit the domain?

**Reply by robotictacos** [reply]
Ah, yes, it pops up *very* quickly but this time I caught it, the error says Domain does not point to this server or reverse proxy not configured correctly.  So, I suppose I have not progressed after all.

**Reply by szaimen** [reply]
ah good point. we should probably increase the timeout

**Reply by szaimen** [reply]
are 5s enough, what do you think?
(currently it is 3s)

**Reply by robotictacos** [reply]
I'd make it 10, personally.  I'd estimate it was actually viewable on my screen for 1-2 seconds.  Fortunately I'm a pretty fast reader but even still I had to click the submit button 3-4 times to copy the text of the error.

**Reply by szaimen** [reply]
okay

**Reply by szaimen** [reply]
okay, 10s it is

**Reply by robotictacos** [reply]
So trying to trace why it doesn't like my domain, I see this entry in the logs:

{"level":"error","ts":1651685786.0907974,"logger":"tls.obtain","msg":"could not get certificate from issuer","identifier":"nextcloud.dazedconfusion.com","issuer":"acme.zerossl.com-v2-DV90","error":"[nextcloud.mydomain.com] solving challenges: presenting for challenge: presenting with embedded solver: could not start listener for challenge server at :80: listen tcp :80: bind: permission denied (order=https://acme.zerossl.com/v2/DV90/order/FneLp_2PET6cj-nYfw88Ug) (ca=https://acme.zerossl.com/v2/DV90)"}

I'm not sure why it would not be able to listen on port 80, I do have it mapped to 16000 (and also forwarded on the firewall, but if I read the error correctly, it's not able to open the port).

**Reply by szaimen** [reply]
sorry, I don't have time to investiage this anymore.
Please reset the whole instance and try from scatch by following this documentation: https://github.com/nextcloud/all-in-one/blob/2c4ba5f4a80e317e318c9938886520869e146a91/reverse-proxy.md (there is now a section for Traefik 2)

**Reply by robotictacos** [reply]
Understand the time constraints.  Even if I follow @macaussie 's example to the letter it doesn't work for me, but I suspect it's due to the different version of Traefik.  Thanks for trying! :)

**Reply by robotictacos** [reply]
I got it!  Also without using TOML file ...

Turns out Apache is running on an undocumented port:
![Capture](https://user-images.githubusercontent.com/47487583/166800232-57a35d9f-f8b3-4eea-b601-08c0a9d0c3b9.PNG)

Once I realized the plain HTTP port was 8000, all I had to do was point Traefik there, here is my final config:


  nextcloud:
    image: nextcloud/all-in-one:latest
    container_name: nextcloud-aio-mastercontainer
    environment:
      - APACHE_PORT=11000
    networks:
      t2_proxy:
        ipv4_address: 172.18.0.16
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config
      - $DOCKERDIR/nextcloud/backup:/mnt/backup
      - $DOCKERDIR/nextcloud/data:/data
      - /var/run/docker.sock:/var/run/docker.sock:ro
    ports:
      - 3478:3478
    restart: unless-stopped
    labels:
      - "traefik.enable=true"
      ## HTTP Routers
      - "traefik.http.routers.nextcloud-rtr.entrypoints=https"
      - "traefik.http.routers.nextcloud-rtr.rule=Host(`nextcloud.$DOMAINNAME`)"
      ## Middlewares
      - "traefik.http.routers.nextcloud-rtr.middlewares=chain-no-auth@file"
      ## HTTP Services
      - "traefik.http.routers.nextcloud-rtr.service=nextcloud-svc"
      - "traefik.http.services.nextcloud-svc.loadbalancer.passHostHeader=true"
      - "traefik.http.services.nextcloud-svc.loadbalancer.server.port=8000"

I left the environment variable to 11000 but it doesn't seem to do anything.  I appreciate your help!!

**Reply by szaimen** [reply]
It seems like you still havent read https://github.com/nextcloud/all-in-one/discussions/543#discussioncomment-2669959 and https://github.com/nextcloud/all-in-one/discussions/553#discussioncomment-2679936 and https://github.com/nextcloud/all-in-one/blob/2c4ba5f4a80e317e318c9938886520869e146a91/reverse-proxy.md

**Reply by szaimen** [reply]
You simply cannot use Docker Labels for this

**Reply by robotictacos** [reply]
Eh, but I am?  It's working at the moment with only labels, using the posted config.  It didn't work for me using the config posted in the reverse proxy readme.

**Reply by szaimen** [reply]
Yes but only for the aio interface. What you want to make it work for is the domainchwck and apache container that get spawned by the mastercontainer

**Reply by szaimen** [reply]
Maybe reading this would already explain a few things: https://github.com/nextcloud/all-in-one#how-does-it-work

**Reply by robotictacos** [reply]
No, I meant Nextcloud is working, I created a user login etc.  I do appreciate your help though.

**Reply by szaimen** [reply]
okay if you made it work like this you are likely a magician.
Are you sure that you did not use the toml file?

**Reply by szaimen** [reply]
can you post the output of `sudo docker inspect nextcloud-aio-apache` please?

**Reply by szaimen** [reply]
Hello, I see we have to drastically improve the reverse proxy documentation. I already have some ideas but will probably not come to it before this afternoon.

**Comment by macaussie** [comment]
@szaimen saw the updated doc. Looks look.

Just an explanation why I personally am mapping the 16000 ports. I have other services running on 80, 8080 & 8443.

**Reply by szaimen** [reply]
Thanks for the feedback! Will merge this then now :)

Yes, thought so that these ports are already in use :)

**Comment by szaimen** [comment, marked as answer]
For anyone reading this, please follow the reworked reverse proxy documentation: https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md
Feedback is welcome!

**Comment by macaussie** [comment]
Hi Simon

I have a further update and shall share my files. Whilst it was working, I was still getting the following errors:

The "X-Robots-Tag" HTTP header is not set to "none". This is a potential security or privacy risk, as it is recommended to adjust this setting accordingly.
The "X-Frame-Options" HTTP header is not set to "SAMEORIGIN". This is a potential security or privacy risk, as it is recommended to adjust this setting accordingly.

What I have done to setup my system is as follows:

docker-compose file:
  nextcloud:
    image: nextcloud/all-in-one:latest
    restart: always
    container_name: nextcloud-aio-mastercontainer
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config
      - /var/run/docker.sock:/var/run/docker.sock:ro
    ports:
      - 16000:8080
    environment:
      - APACHE_PORT=11000

volumes:
  nextcloud_aio_mastercontainer:
    name: nextcloud_aio_mastercontainer


Traefik/rules/nc.toml file:

[http.routers]
    [http.routers.nc-rtr]
        entryPoints = ["https"]
        rule = "Host(`my.domain.com`)"
        service = "nc-svc"
        middlewares = ["chain-nc"]
        [http.routers.nc-rtr.tls]
            certresolver = "dns-cloudflare"

[http.services]
    [http.services.nc-svc]
        [http.services.nc-svc.loadBalancer]
            passHostHeader = true
            [[http.services.nc-svc.loadBalancer.servers]]
                url = "http://xxx.xxx.xxx.xxx:11000"



Traefik/rules/middlewares.toml file:

#added at the bottom
  [http.middlewares.nc-middlewares-secure-headers]
    [http.middlewares.nc-middlewares-secure-headers.headers]
      accessControlMaxAge = 100
      hostsProxyHeaders = ["X-Forwarded-Host"]
      sslRedirect = true
      stsSeconds = 63072000
      stsIncludeSubdomains = true
      stsPreload = true
      forceSTSHeader = true
      customFrameOptionsValue = "SAMEORIGIN"
      contentTypeNosniff = true
      browserXssFilter = true
      referrerPolicy = "same-origin"
      featurePolicy = "camera 'none'; geolocation 'none'; microphone 'none'; payment 'none'; usb 'none'; vr 'none';"
      [http.middlewares.nc-middlewares-secure-headers.headers.customResponseHeaders]
        X-Robots-Tag = "none"
        server = ""


Traefik/rules/middleware-chains.toml file

# added at the bottom

  [http.middlewares.chain-nc]
    [http.middlewares.chain-nc.chain]
      middlewares = [ "middlewares-rate-limit", "nc-middlewares-secure-headers"]


No more errors and an A+ on the security scan.

Cheers,
James

**Reply by szaimen** [reply]
Hi,

thanks for this but there appears to be something wrong. You should not need to set all the headers manually as Nextcloud should provide them on its own. The only header needed should be the HSTS setting and maybe also the x-forwarded-host headder. Can you please investigate this?

Thank you!

**Reply by szaimen** [reply]
So basically I think these should be enough?

```
[http.middlewares.nc-middlewares-secure-headers]
[http.middlewares.nc-middlewares-secure-headers.headers]
hostsProxyHeaders = ["X-Forwarded-Host"]
sslRedirect = true
stsSeconds = 63072000
stsIncludeSubdomains = true
stsPreload = true
forceSTSHeader = true
```

**Comment by macaussie** [comment]
All of those and these two

referrerPolicy = "same-origin"
X-Robots-Tag = "none"

I assume this is because NextCloud is behind the reverse proxy.


It works without any of this but I do get the warnings with I look at overview.

**Reply by szaimen** [reply]
okay, I'll create a PR then

**Reply by szaimen** [reply]
thanks for investigating!

**Reply by macaussie** [reply]
Great job on this though. Really impressed with it.

**Reply by szaimen** [reply]
Thanks for the positive feedback! :)

Thank you for helping getting the reverse proxy documentation for Traefik right! :)

**Reply by szaimen** [reply]
@macaussie does this look good to you? :) See https://github.com/nextcloud/all-in-one/pull/592

You can get a preview of this here: https://github.com/nextcloud/all-in-one/blob/enh/543/treafik/reverse-proxy.md#traefik-2

**Reply by macaussie** [reply]
Yeah that looks good.

**Reply by szaimen** [reply]
Thank you! Will merge then :)

**Reply by szaimen** [reply]
Short question: can everything of this not be in one file or do these have to be different files?

**Reply by macaussie** [reply]
I believe they have to be seperate files.

**Reply by szaimen** [reply]
Okay, then lets leave it like this :)

**Comment by chrisbley** [comment]
I keep getting the "The server is not reachable on port 443" error on the AIO setup page being behind Traefik.
I configured the AIO container using docker labels; Traefik dashboard says that everything is ok, from entrypoint 443 to Service "nextcloud-aio-mastercontainer". I've used my portforwardings for ports 80 and 443 successfully with Nginx Proxy Manager in the past, so I don't think there's anything wrong with my port 443. Any advice how to debug would be very much appreciated - Cheers!

**Reply by szaimen** [reply]
See https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md

**Comment by robotictacos** [comment]
You have to access the AIO page using port 8080, then you can hit the main Nextcloud page via 443.
  
  
  
  
  
  
  

  
  
>

**Comment by chrisbley** [comment]
That's exactly what I cannot do - I am on the AIO setup page via my local
8181:8080 port mapping, and getting that port 443 error when putting in my
domain name.

Sent on the go.

Dave Campbell ***@***.***> schrieb am Di., 10. Mai 2022,
14:22:

>
>
>
> You have to access the AIO page using port 8080, then you can hit the main
> Nextcloud page via 443.
>
>
>
>
>
>
>
>
>
>
> >

**Reply by szaimen** [reply]
See https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md

**Comment by robotictacos** [comment]
Yeah you will get that if the reverse proxy isn’t set correctly.    I also had that issue.    If you are on traefik (I assume you are since you are replying to this discussion) you have to set the url to http://yourip:11000 in your toml file
  
  
  
  
  
  
  

  
  
>

**Comment by chrisbley** [comment]
thank you for your replies, @robotictacos! Had to make a break as this was eating too much of my time.

I still haven't managed to get it working - this is my traefik config, after translating szaimen's reverse proxy guide to my yaml file:
https://www.toptal.com/developers/hastebin/nucojotuno.yaml

could you please have a quick look at it, maybe you can find the mistake?

thank you!

**Reply by robotictacos** [reply]
Hi Chris,
Sad to say, I gave up on AIO.  I did manage to get it to work, but then if you need to tweak something inside the container, like for me a path to a folder, it's kind of impossible the way it's currently implemented.  Also, I despise volume mapping, I pretty much never ever do that simply because I want access to the folders being mounted inside the guest, and this implementation is absolutely riddled with volumes.  It was quite a time sink for not much payoff.  If you like I can show my yaml with a more conventional configuration which is performing nicely.

**Reply by szaimen** [reply]
Regarding being able to tweak things, did you see this https://github.com/nextcloud/all-in-one/tree/main/manual-install?

**Reply by robotictacos** [reply]
No, that wasn't an option when I was working on this previously, that does seem to reflect what I wanted to do though.

**Reply by chrisbley** [reply]
hey @robotictacos , thank you for your feedback; I'd love to see your configuration that worked for you!
(I'm still wondering if I should generally switch to toml config instead of yaml config, as most container documentation seem to have toml but not yaml config examples...)
Cheers!

**Comment by chrisbley** [comment]
Hey @robotictacos, seems my reply didn't get through last time. If you don't mind,
could you send me your working traefik config?
Thanks a lot!

Sent on the go.

Dave Campbell ***@***.***> schrieb am Di., 24. Mai 2022,
01:40:

**Reply by jonnyparris** [reply]
Hey @chrisbley - did you get this working in the end? I'm also getting the 443 error even though https://portchecker.co/check says that the port is open 🤔 . Does that perhaps resonate with any other error / misconfiguration that you've encountered @szaimen ?
(and yes I tried the docs version first, but it didn't work for me, but using labels did).
I also tried using Caddy too but it complains that the 443 has already been bound because I've got Traefik runnning fine for a bunch of other services on the same server. I'm open to ideas to get them playing nicely together failing an update/tweak that gets Traefik working first.
Thanks in advance!

**Comment by chrisbley** [comment]
Hey Ruskin,

No I didn't get it working! It appears to be not about open ports, but
rather about headers.
I gave up on it.

Sent on the go.

Ruskin ***@***.***> schrieb am Mi., 17. Aug. 2022, 23:43:

**Reply by jonnyparris** [reply]
Ah shame. Thanks for the headers hint; I might dive deeper.
Did you give up on Traefik or AIO or both?

**Comment by chrisbley** [comment]
I gave up on both. Maybe Traefik is just too complicated for me, but 
there was several things that I failed to do with Trafik: safely expose 
the API, configure header forwarding for any installation of Nextcloud 
(docker, VM etc.) and other things.
I'm going to use the Nextcloud VM by Hansson and pay them for making it 
work behind my Nginx Proxy Manager. I've dumped so much time in this 
already that the 60 Euro per 30min for their service look like good 
value to me.


------ Originalnachricht ------
Von "Ruskin" ***@***.***>
An "nextcloud/all-in-one" ***@***.***>
Cc "chrisbley" ***@***.***>; "Mention" 
***@***.***>
Datum 22.08.2022 13:15:09
Betreff Re: [nextcloud/all-in-one] Running behind Traefik Reverse Proxy 
(Discussion #543)

>
>Ah shame. Thanks for the headers hint; I might dive deeper.
>Did you give up on Traefik or AIO or both?
>
>—
>Reply to this email directly, view it on GitHub 
><https://github.com/nextcloud/all-in-one/discussions/543#discussioncomment-3446261>, 
>or unsubscribe 
><https://github.com/notifications/unsubscribe-auth/AXKZ5OCYAEPWXVFLWCL4N6DV2NOL3ANCNFSM5U2BWWYA>.
>You are receiving this because you were mentioned.Message ID: 
***@***.***>
>

**Reply by jonnyparris** [reply]
fwiw I just abandoned traefik completely and used cloudflare tunnels instead and managed to get things working _finally_.

Disclaimer: I work for CF, but not on the tunnels product and it's my first time using it as a reverse proxy like this. Pleasantly surprising all the same after sinking so many hours into wrestling with Traefik 😤
